metals/src/main/scala/scala/meta/internal/metals/debug/BuildTargetClasses.scala

```diff
@@ -271,6 +316,297 @@ final class BuildTargetClasses(val buildTargets: BuildTargets)(implicit
+                }
+              testClasses += ((classSymbolWithoutFunctionName, testInfo))
+            case None =>
+          }
+        case _ =>
+      }
+    }
+  }
+
+  private def processTestClassHierarchy(
+      symbolInfo: SymbolInformation,
+      doc: TextDocument,
+      path: AbsolutePath,
+      testClasses: scala.collection.mutable.ListBuffer[(String, TestSymbolInfo)],
+  ): Option[Future[Unit]] = {
+    symbolInfo.signature match {
+      case classSig: ClassSignature =>
+        val symbol = symbolInfo.symbol
+        val className = symbolToClassName(symbol)
+        if (className.nonEmpty) {
+          Some(
+            detectTestFrameworkUsingClassHierarchy(classSig, doc, path)
+              .map { frameworkOpt =>
+                frameworkOpt.foreach { framework =>
+                  val testInfo = TestSymbolInfo(className, framework)
+                  testClasses += ((symbol, testInfo))
+                }
+              }
+          )
+        } else None
+      case _ => None
+    }
+  }
+
+  private def detectTestFrameworkUsingClassHierarchy(
+      classSig: ClassSignature,
+      doc: TextDocument,
+      path: AbsolutePath,
+  ): Future[Option[TestFramework]] = {
+    val initialParents = extractParentSymbols(classSig)
+    searchClassHierarchyForTestFramework(
+      initialParents,
+      doc,
+      path,
+    )
+  }
+
+  private def searchClassHierarchyForTestFramework(
+      symbols: List[String],
+      doc: TextDocument,
+      path: AbsolutePath,
+      visited: Set[String] = Set.empty,
+  ): Future[Option[TestFramework]] = {
+    if (symbols.isEmpty) {
+      Future.successful(None)
+    } else {
+      val directFramework = symbols
+        .map(symbol =>
+          TestFrameworkDetector
+            .fromSymbol(symbol)
+        )
+        .collectFirst { case Some(framework) => framework }
+
+      directFramework match {
+        case Some(framework) => Future.successful(Some(framework))
+        case None =>
+          val nextLevelFutures = symbols
+            .filterNot(visited.contains)
+            .map(symbol => collectParentsForSymbol(symbol, doc, path, visited))
+          Future.sequence(nextLevelFutures).flatMap { parentList =>
+            val allParents = parentList.flatten.distinct
+
+            searchClassHierarchyForTestFramework(
+              allParents,
+              doc,
+              path,
+              visited ++ symbols,
+            )
+          }
+      }
+    }
+  }
+
+  private def extractParentSymbols(classSig: ClassSignature): List[String] = {
+    classSig.parents.collect { case TypeRef(_, parentSymbol, _) =>
+      parentSymbol
+    }.toList
+  }
+
+  private def fetchAndCacheSymbolInfo(
+      path: AbsolutePath,
+      symbol: String,
+  ): Future[Option[PcSymbolInformation]] = {
+    compilers().info(path, symbol).map { result =>
+      symbolInfoCache.put(symbol, (result, path))
+      result
+    }
```

Comment:
```suggestion
    compilers().info(path, symbol).flatMap { result =>
      symbolIndex.definition(mtags.Symbol(symbol)).map{ definition =>
        symbolInfoCache.put(symbol, (result, definition.path))
        symbolDefinitionCache.updateWith( definition.path){
          added => 
           add.getOrElse(Set.empty) + symbol
        } 
        result
      }
    }
```
something along this lines would help use keep information where the base suites are defined and invalidate properly